File: tasks/2024/round3_pp/peaks-pp.py
#!/usr/bin/env python3
# @check-accepted: examples N1 NMsmall no-limits
# insert brief description of the solution here


# input data
N,M = list(map(int, input().strip().split()))
A = list(map(int, input().strip().split()))
B = list(map(int, input().strip().split()))

peaksA = 0
peaksB = 0

for i in range(N):
    if (i == 0 or A[i] > A[i-1]) and (i == N-1 or A[i] > A[i+1]):
        peaksA += 1

for i in range(M):
    if (i == 0 or B[i] > B[i-1]) and (i == M-1 or B[i] > B[i+1]):
        peaksB += 1

print(peaksA * peaksB)  

File: tasks/2024/round3_pp/peaks-pp.cpp
// @check-accepted: examples N1 NMsmall no-limits
// insert brief description of the solution here

#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

int main() {

    int N, M;
    cin >> N >> M;
    vector<int> A(N), B(M);
    for(int &x: A) cin >> x;
    for(int &x: B) cin >> x;

    long long peaksA = 0, peaksB = 0;

    for(int i = 0; i < N; ++i)
        if((i == 0 || A[i] > A[i-1]) && (i == N-1 || A[i] > A[i+1]))
            ++peaksA;
    for(int i = 0; i < M; ++i)
        if((i == 0 || B[i] > B[i-1]) && (i == M-1 || B[i] > B[i+1]))
            ++peaksB;

    cout << peaksA * peaksB << endl;

    return 0;
}
